python/lsst/cp/pipe/defects.py

```diff
@@ -148,472 +114,104 @@ class FindDefectsTaskConfig(pexConfig.Config):
-            return []
-
-        if len(defectList) == 1:  # single input - no merging to do
-            return defectList[0]
-
-        sumImage = afwImage.MaskedImageF(imageDimensions)
-        for defects in defectList:
-            for defect in defects:
-                sumImage.image[defect.getBBox()] += 1
-        sumImage /= len(defectList)
-
-        nDetected = len(np.where(sumImage.image.array > 0)[0])
-        self.log.info("Pre-merge %s pixels with non-zero detections" % nDetected)
-
-        if mode == 'OR':  # must appear in any
-            indices = np.where(sumImage.image.array > 0)
-        else:
-            if mode == 'AND':  # must appear in all
-                threshold = 1
-            elif mode == 'FRACTION':
-                threshold = self.config.combinationFraction
-            else:
-                raise RuntimeError(f"Got unsupported combinationMode {mode}")
-            indices = np.where(sumImage.image.array >= threshold)
-
-        BADBIT = sumImage.mask.getPlaneBitMask('BAD')
-        sumImage.mask.array[indices] |= BADBIT
-
-        self.log.info("Post-merge %s pixels marked as defects" % len(indices[0]))
-
-        if self.config.edgesAsDefects:
-            self.log.info("Masking edge pixels as defects in addition to previously identified defects")
-            self._setEdgeBits(sumImage, 'BAD')
-
-        defects = measAlg.Defects.fromMask(sumImage, 'BAD')
-        return defects
-
-    @staticmethod
-    def _getNumGoodPixels(maskedIm, badMaskString="NO_DATA"):
-        """Return the number of non-bad pixels in the image."""
-        nPixels = maskedIm.mask.array.size
-        nBad = countMaskedPixels(maskedIm, badMaskString)
-        return nPixels - nBad
+        nPix = 0
+        for defect in defects:
+            nPix += defect.getBBox().getArea()
+        return nPix
 
-    def findHotAndColdPixels(self, exp, imageType, setMask=False):
+    def findHotAndColdPixels(self, exp, nSigma):
         """Find hot and cold pixels in an image.
 
-        Using config-defined thresholds on a per-amp basis, mask pixels
-        that are nSigma above threshold in dark frames (hot pixels),
-        or nSigma away from the clipped mean in flats (hot & cold pixels).
+        Using config-defined thresholds on a per-amp basis, mask
+        pixels that are nSigma above threshold in dark frames (hot
+        pixels), or nSigma away from the clipped mean in flats (hot &
+        cold pixels).
 
         Parameters
         ----------
         exp : `lsst.afw.image.exposure.Exposure`
             The exposure in which to find defects.
-        imageType : `str`
-            The image type, either 'dark' or 'flat'.
-        setMask : `bool`
-            If true, update exp with hot and cold pixels.
-            hot: DETECTED
-            cold: DETECTED_NEGATIVE
+        nSigma : `list [ `float` ]
+            Detection threshold to use.  Positive for DETECTED pixels,
+            negative for DETECTED_NEGATIVE pixels.
```

Comment:
It probably is only useful in the case that the length is less than or equal to 2.  As noted above, I've added a case for DARK inputs frames.  I agree that longer lists will be redundant, but this pulls logic out of the `findHotAndColdPixels` method and puts it in `run`, so that method only needs to handle the thresholds given.  This seemed clearer to me, as from the `run` method, it's obvious what the `find` method will do.